gui/gui.spec

```diff
@@ -1,76 +1,92 @@
 # -*- mode: python -*-
 import sys
-from PyInstaller.utils.hooks import (collect_all, collect_submodules)
+from PyInstaller.utils.hooks import collect_all, collect_submodules
 
 IS_MAC = sys.platform == "darwin"
 
 block_cipher = None
 
 datas, binaries, hiddenimports = [], [], []
-for pkgname in ['glean', 'glean_parser', 'mozregression', 'yamllint', 'bs4']:
+for pkgname in ["glean", "glean_parser", "mozregression", "yamllint", "bs4"]:
     pkg_datas, pkg_binaries, pkg_hiddenimports = collect_all(pkgname)
     datas.extend(pkg_datas)
     binaries.extend(pkg_binaries)
     hiddenimports.extend(pkg_hiddenimports)
 
-# workaround this bad interaction between setuptools and pyinstaller:
-# https://github.com/pypa/setuptools/issues/1963
-hiddenimports.extend(collect_submodules('pkg_resources'))
+a = Analysis(
+    ["mozregression-gui.py"],
+    binaries=binaries,
+    datas=datas,
+    hiddenimports=hiddenimports,
+    hookspath=[],
+    runtime_hooks=[],
+    excludes=[],
+    win_no_prefer_redirects=False,
+    win_private_assemblies=False,
+    cipher=block_cipher,
+    noarchive=False,
+)
 
-a = Analysis(['mozregression-gui.py'],
-             pathex=['/Users/wlach/src/mozregression/gui'],
-             binaries=binaries,
-             datas=datas,
-             hiddenimports=hiddenimports,
-             hookspath=[],
-             runtime_hooks=[],
-             excludes=[],
-             win_no_prefer_redirects=False,
-             win_private_assemblies=False,
-             cipher=block_cipher,
-             noarchive=False)
 pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)
 
 if IS_MAC:
-    exe = EXE(pyz,
-            a.scripts,
-            a.binaries,
-            a.zipfiles,
-            a.datas,
-            [],
-            name='mozregression GUI',
-            debug=False,
-            bootloader_ignore_signals=False,
-            strip=False,
-            upx=False,
-            console=False,
-            target_arch="universal2")
-    app = BUNDLE(exe,
-                strip=False,
-                upx=True,
-                name='mozregression GUI.app',
-                icon='icons/app_icon.icns',
-                bundle_identifier=None,
-                info_plist={
-                    'NSPrincipalClass': 'NSApplication',
-                    'NSHighResolutionCapable': 'True'
-                })
+    binaries = a.binaries - TOC(
```

Comment:
Please add some comments describing what is going on here, or move this into a well-named function.